changelog/11367.bugfix.rst

```diff
@@ -0,0 +1,4 @@
+[junit] move properties handling to :func:`finalize()`
+
+seem like not all the cases of calling finalize were adding the properties into the test case element,
+which leads to some test cases missing the expected properties.
```

Comment:
The entry should be written in a user-facing language instead of the internal implementation details. 👍 

```suggestion
Fixed bug where `user_properties` where not being saved in the JUnit XML file if a fixture failed during teardown.
```

Reply:
Sorry about that, I'll try better next time 😬

Reply:
No worries, that's why reviewers are for. 😁 

Thanks again!